collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2794,15 +2802,19 @@ public void onNumberPickerValueSelected(int widgetId, int value) {
     public void onDateChanged(LocalDateTime date) {
```

Comment:
Both `onDateChanged()` and `onRankingChanged()` have the same body so probably we can improve - move the body to a separate method or even use just one listener for both.

Reply:
Agreed. Do you want this done here? I feel like there's already a lot of random stuff going on in this PR and it may be better to separate that out so there can be a discussion on the approach.

Reply:
It can be in a follow-up pr, just added a comment to keep it in mind.